# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/base/nsFocusManager.cpp
# Commit: 6ee65cf29009
# Full Hash: 6ee65cf29009a22701f733a039d0578cc05cf75d
# Author: Edgar Chen <echen@mozilla.com>
# Date: 2022-03-22 04:27:58
# Description:
#   Bug 1756233 - Ensure docShellBeingHidden is not null; r=hsivonen
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D141515
# ==============================================================================

diff -r 4c28f1640878 -r 6ee65cf29009 dom/base/nsFocusManager.cpp
--- a/dom/base/nsFocusManager.cpp	Mon Mar 21 09:21:40 2022 +0000
+++ b/dom/base/nsFocusManager.cpp	Mon Mar 21 09:58:00 2022 +0000
@@ -1079,7 +1079,8 @@
   // need to consider process switches where the hiding docshell is already
   // remote (ie. GetEmbedderElement is nullptr), as shifting remoteness to the
   // frame element is handled elsewhere.
-  if (nsDocShell::Cast(docShellBeingHidden)->WillChangeProcess() &&
+  if (docShellBeingHidden &&
+      nsDocShell::Cast(docShellBeingHidden)->WillChangeProcess() &&
       docShellBeingHidden->GetBrowsingContext()->GetEmbedderElement()) {
     if (mFocusedWindow != window) {
       // The window being hidden is an ancestor of the focused window.
